### PR TITLE
RE-1425 RPC Gating Container Options for venv

### PR DESCRIPTION
**Pull request by dealako**
- Moved tar and ssh to inside the container
- Added ssh agent support

Issue: [RE-1425](https://rpc-openstack.atlassian.net/browse/RE-1425)

### CODE REVIEW

**Comment by odyssey4me** [comment]
recheck_cit_all

**Comment by odyssey4me** [comment]
recheck_issue_link

File: rpc_jobs/build_gating_venv.yml [hunk 1]
```diff
@@ -20,102 +20,99 @@
...
-                                        'git@github.com:rcbops/rpc-gating',
-                                        env.RPC_GATING_BRANCH)
-              dir('rpc-gating'){
-                // run in docker to ensure all the required apt packages are available
-                container = docker.build env.BUILD_TAG.toLowerCase()
-              }
-              // Venv is build within docker as it requires extra packages
-              container.inside{
-                sh """#!/bin/bash -xeu
+              withCredentials(artifact_build.get_rpc_repo_creds()) {
```
Comment (git-harry) [inline rpc_jobs/build_gating_venv.yml]: Is this needed by anything other than the shell script? If not, I'd say only wrap it around what's required.

**Comment by hughsaunders** [comment]
recheck_issue_link

**Comment by hughsaunders** [comment]
recheck_issue_link

**Comment by hughsaunders** [comment]
recheck_issue_link

**Comment by dealako** [comment]
recheck_all

File: rpc_jobs/build_gating_venv.yml [hunk 2]
```diff
@@ -20,102 +20,99 @@
...
+                }
 
-                # Install virtualenv if required
-                if [[ ! -d ".venv" ]]; then
-                  requirements="virtualenv==15.1.0"
-                  pip install -U "\${requirements}" \
-                    || pip install --isolated -U "\${requirements}"
+                // Venv is built within docker, includes all the environment variables
+                container.inside{
+                  sshagent(credentials:['rpc_repo_ssh_user_private_key']){
```
Comment (mattt416) [inline rpc_jobs/build_gating_venv.yml]: Friday afternoon brain here, but I don't see this credential.  Should this be `RPC_REPO_SSH_USER_PRIVATE_KEY_FILE`?

File: rpc_jobs/build_gating_venv.yml (hunk 1 above)
Comment (mattt416) [reply rpc_jobs/build_gating_venv.yml]: There's a note that says:

```
// venv is pushed to rpc repo from outside docker container, as binding
// credentials files does not work within docker (used for ssh key)
```

... which explains why this has to wrap the container.

File: rpc_jobs/build_gating_venv.yml (hunk 2 above)
Comment (mattt416) [reply rpc_jobs/build_gating_venv.yml]: Similar to @git-harry's comment above, I'd probably only wrap the ssh commands in the sshagent block to reduce the time the agent's running.

**Comment by dealako** [comment]
recheck_unit_tests

File: rpc_jobs/build_gating_venv.yml [hunk 3]
```diff
@@ -17,108 +17,120 @@
...
+                  # Determine the SHA and archive name that we'll use
+                  pushd rpc-gating; SHA=\$(git rev-parse HEAD); popd
+                  archive="rpcgatingvenv_\${SHA}.tbz"
+
+                  # Set some SSH options
+                  SSH_OPTS="-o StrictHostKeyChecking=no"
+
+                  # Check to see if the virtual env archive already exists...if so, no need to rebuild it
+                  ssh \$SSH_OPTS \$REPO_USER@\$REPO_HOST "test -e \$REPO_PATH/\$archive"
+                  if [[ $? -eq 0 ]]; then
```
Comment (hughsaunders) [inline rpc_jobs/build_gating_venv.yml]: `$?` needs escaping as otherwise groovy will interpret that `$`. This is causing the unit test fail.

**Comment by dealako** [comment]
Closed in favor of https://github.com/rcbops/rpc-gating/pull/804

**Pull request closed without merging**